compiler/rustc_hir_typeck/src/expr_use_visitor.rs

```diff
@@ -80,21 +89,138 @@ pub trait Delegate<'tcx> {
     );
 }
 
-#[derive(Copy, Clone, PartialEq, Debug)]
-enum ConsumeMode {
-    /// reference to x where x has a type that copies
-    Copy,
-    /// reference to x where x has a type that moves
-    Move,
+impl<'tcx, D: Delegate<'tcx>> Delegate<'tcx> for &mut D {
```

Comment:
All call sites pass in `&mut Delegate` but I don't want to store a lifetime in the `ExprUseVisitor` struct lol